apps/wallet-mobile/src/Staking/PoolTransition/usePoolTransition.tsx

```diff
@@ -0,0 +1,426 @@
+import {PoolInfo} from '@emurgo/yoroi-lib'
+import {useNavigation} from '@react-navigation/native'
+import {useTheme} from '@yoroi/theme'
+import {Balance} from '@yoroi/types'
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {Image, StyleSheet, Text, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import {useQuery} from 'react-query'
+
+import {Button, Icon, Spacer, useModal} from '../../components'
+import {useStakingInfo} from '../../Dashboard/StakePoolInfos'
+import {useSelectedWallet} from '../../features/WalletManager/Context'
+import {formatTimeSpan, Quantities} from '../../yoroi-wallets/utils'
+
+// TODO: PoolTransition and getPoolTransition will come from yoroi-lib
+type PoolTransition = {
+  currentPool: PoolInfo & {tax: number}
+  suggestedPool: PoolInfo & {tax: number}
+  deadlineMilliseconds: number
+}
+
+const getPoolTransition = async (currentPoolId: string): Promise<PoolTransition | null> => {
```

Comment:
The function was defined by @vsubhuman and is to be implemented in yoroi-lib

Reply:
Also, this doesn't get the PoolInfo, it gets the PoolTransition, which indicate current PoolInfo, suggested PoolInfo and deadline till current pool is expired.

Reply:
it doesn't make sense to me to have 2 different endpoints for that, whenever the pool info is requested it should return if the pool is retiring or not cc: @vsubhuman

Reply:
This would be ideal, separation of concerns, and not a big soup.

Reply:
The new pool suggestion can be done in the same way of FFEE, S3 with cloud front on top, it will fill the needs

Reply:
I assumed yoroi-lib getting the deadlines and suggested pool would be via an S3 file or similar, yes.

But right now yoroi-lib gets PoolInfo from `cexplorer.io`.

You're saying when we transform PoolInfo from the api, we add there the deadline and suggested pool from the S3 file there as well?

Reply:
it looks like there is an API for the pool info coming from the aggregator, need to check with @pedromtcosta

Reply:
@stackchain we can return the epoch of the Pool retirement certificate, if that's what you are referring to. Should be an easy change. For reference: https://github.com/IntersectMBO/cardano-ledger/blob/8884d921c8c3c6e216a659fca46caf729282058b/eras/babbage/test-suite/cddl-files/babbage.cddl#L187

Reply:
I'll ask, but I'm not sure the onchain pool retirement is what management are looking for. It's a good information to have, specially for other pools. 
But they have a specific list, and another list to migrate to. And they may want to specify an earlier date than the actual retirement, so users have time to migrate without loosing rewards.

Reply:
@pedromtcosta @jorbuedo when creating the retirement certificate it can be created with an specific epoch ahead, so since is current_epoch + 1, it can be planned ahead, and API can return it, it would be nice to receive the retirement `epoch` and `date` (beginning of epoch) from the API so the FE doesn't need to do any math to get the countdown. Also @pedromtcosta the pool can be registered again, which means API need to handle retirement->registration->retirement->registration. (ty: @vsubhuman)

Reply:
That's good info to have, but doesn't solve the current request to have a list of retiring pools and a list of suggested pools to transition to. We may not even have access to the retiring pools to add the certificate.

Reply:
The list of retiring pools is no longer required with this implementation, whenever a pool the user is delegating is about to be retired the pop will show, the only thing needed is the suggestion (transition endpoint), in this scenario.

Reply:
We had a call, decided to go with current approach just for now, based on this API https://github.com/Emurgo/yoroi-lib/pull/136

Any future decisions on this, API improvements and support for other pools will be followed up separately.